src/BenchmarksApps/TodosApi/DatabaseInitializer.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.Extensions.Options;
+using Npgsql;
+
+namespace TodosApi;
+
+internal class DatabaseInitializer : IHostedService
+{
+    private readonly NpgsqlDataSource _db;
+    private readonly ILogger<DatabaseInitializer> _logger;
+    private readonly bool _initDatabase;
+
+    public DatabaseInitializer(NpgsqlDataSource db, IOptions<AppSettings> appSettings, ILogger<DatabaseInitializer> logger)
+    {
+        _db = db;
+        _logger = logger;
+        _initDatabase = !appSettings.Value.SuppressDbInitialization;
+    }
+
+    public Task StartAsync(CancellationToken cancellationToken)
+    {
+        if (_initDatabase)
+        {
+            return Initialize(cancellationToken);
+        }
+
+        if (_logger.IsEnabled(LogLevel.Information))
+        {
+            _logger.LogInformation("Database initialization disabled for connection string '{connectionString}'", _db?.ConnectionString);
+        }
+
+        return Task.CompletedTask;
+    }
+
+    public Task StopAsync(CancellationToken cancellationToken) => Task.CompletedTask;
+
+    private async Task Initialize(CancellationToken cancellationToken = default)
+    {
+        // NOTE: Npgsql removes the password from the connection string
+        if (_logger.IsEnabled(LogLevel.Information))
+        {
+            _logger.LogInformation("Ensuring database exists and is up to date at connection string '{connectionString}'", _db.ConnectionString);
+        }
```

Comment:
This doesn't seem like a good thing to log in general...

Reply:
This used to include code that sanitized the connection string but it turns out Npgsql does that already. But as you say it might be better to just not log it at all.